File: src/runtime/node/node_http_binding.zig
pub fn getBunServerAllClosedPromise(globalThis: *jsc.JSGlobalObject, callframe: *jsc.CallFrame) bun.JSError!jsc.JSValue {
    const arguments = callframe.arguments_old(1).slice();
    if (arguments.len < 1) {
        return globalThis.throwNotEnoughArguments("getBunServerAllClosePromise", 1, arguments.len);
    }

    const value = arguments[0];

    inline for ([_]type{
        jsc.API.HTTPServer,
        jsc.API.HTTPSServer,
        jsc.API.DebugHTTPServer,
        jsc.API.DebugHTTPSServer,
    }) |Server| {
        if (value.as(Server)) |server| {
            return server.getAllClosedPromise(globalThis);
        }
    }

    return globalThis.throwInvalidArgumentTypeValue("server", "bun.Server", value);
}

pub fn getMaxHTTPHeaderSize(globalThis: *jsc.JSGlobalObject, callframe: *jsc.CallFrame) bun.JSError!jsc.JSValue {
    _ = globalThis; // autofix
    _ = callframe; // autofix
    return jsc.JSValue.jsNumber(bun.http.max_http_header_size);
}

pub fn setMaxHTTPHeaderSize(globalThis: *jsc.JSGlobalObject, callframe: *jsc.CallFrame) bun.JSError!jsc.JSValue {
    const arguments = callframe.arguments_old(1).slice();
    if (arguments.len < 1) {
        return globalThis.throwNotEnoughArguments("setMaxHTTPHeaderSize", 1, arguments.len);
    }
    const value = arguments[0];
    const num = try value.coerceToInt64(globalThis);
    if (num <= 0) {
        return globalThis.throwInvalidArgumentTypeValue("maxHeaderSize", "non-negative integer", value);
    }
    bun.http.max_http_header_size = @intCast(num);
    return jsc.JSValue.jsNumber(bun.http.max_http_header_size);
}

const bun = @import("bun");
const jsc = bun.jsc;

File: src/runtime/node/node_http_binding.rs
//! `node:http` native binding — `{get,set}MaxHTTPHeaderSize`.

use bun_jsc::{CallFrame, JSGlobalObject, JSValue, JsResult};

pub(crate) fn get_max_http_header_size(
    _global: &JSGlobalObject,
    _frame: &CallFrame,
) -> JsResult<JSValue> {
    Ok(JSValue::from(bun_http::max_http_header_size()))
}

pub(crate) fn set_max_http_header_size(
    global: &JSGlobalObject,
    frame: &CallFrame,
) -> JsResult<JSValue> {
    let arguments = frame.arguments_old::<1>();
    let arguments = arguments.slice();
    if arguments.is_empty() {
        return Err(global.throw_not_enough_arguments("setMaxHTTPHeaderSize", 1, arguments.len()));
    }
    let value = arguments[0];
    let num = value.coerce_to_int64(global)?;
    if num <= 0 {
        return Err(global.throw_invalid_argument_type_value(
            "maxHeaderSize",
            "non-negative integer",
            value,
        ));
    }
    bun_http::set_max_http_header_size(num as usize);
    Ok(JSValue::from(bun_http::max_http_header_size()))
}
